graphql/util.go

```diff
@@ -1224,3 +1224,76 @@ func collapseCommit(ctx context.Context, mainlineCommits MainlineCommits, mainli
 		mainlineCommitVersion.RolledUpVersions = []*restModel.APIVersion{&apiVersion}
 	}
 }
+
+type PermissionLevel struct {
+	Permission string
+	Level      int
+}
+
+// projectPermissionAccessMap maps project permissions and access levels used in GraphQL directives to Evergreen permission levels.
+var projectPermissionAccessMap = map[ProjectPermission]map[AccessLevel]PermissionLevel{
+	ProjectPermissionSettings: {
+		AccessLevelEdit: {Permission: evergreen.PermissionProjectSettings, Level: evergreen.ProjectSettingsEdit.Value},
+		AccessLevelView: {Permission: evergreen.PermissionProjectSettings, Level: evergreen.ProjectSettingsView.Value},
+	},
+	ProjectPermissionTasks: {
+		AccessLevelAdmin: {Permission: evergreen.PermissionTasks, Level: evergreen.TasksAdmin.Value},
+		AccessLevelEdit:  {Permission: evergreen.PermissionTasks, Level: evergreen.TasksBasic.Value},
+		AccessLevelView:  {Permission: evergreen.PermissionTasks, Level: evergreen.TasksView.Value},
+	},
+	ProjectPermissionAnnotations: {
+		AccessLevelEdit: {Permission: evergreen.PermissionAnnotations, Level: evergreen.AnnotationsModify.Value},
+		AccessLevelView: {Permission: evergreen.PermissionAnnotations, Level: evergreen.AnnotationsView.Value},
+	},
+	ProjectPermissionPatches: {
+		AccessLevelAdmin: {Permission: evergreen.PermissionPatches, Level: evergreen.PatchSubmitAdmin.Value},
+		AccessLevelEdit:  {Permission: evergreen.PermissionPatches, Level: evergreen.PatchSubmit.Value},
+	},
+	ProjectPermissionLogs: {
+		AccessLevelView: {Permission: evergreen.PermissionLogs, Level: evergreen.LogsView.Value},
+	},
+}
+
+const (
+	projectIdKey  = "projectId"
+	identifierKey = "identifier"
+
+	// Keys that are used in the paramsMap.
+	projectIdentifierKey = "projectIdentifier"
```

Comment:
I like the idea of making these constants! Would we consider moving the mapping function into rest/data as well so that we could keep these private keys, but use them across all the params functions?